spec/services/participants/change_schedule_spec.rb

```diff
@@ -0,0 +1,300 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Participants::ChangeSchedule do
+  let(:cohort) { create(:cohort, :current) }
+  let(:lead_provider) { create(:lead_provider) }
+  let(:course) { create(:course, :sl) }
+  let(:course_identifier) { course.identifier }
+  let(:schedule) { create(:schedule, :npq_leadership_spring, cohort:) }
+  let!(:application) { create(:application, :accepted, cohort:, lead_provider:, course:, schedule:) }
+
+  let(:participant) { application.user }
+  let(:participant_id) { participant.ecf_id }
+
+  let(:new_cohort) { create(:cohort, :next) }
+  let(:new_schedule) { create(:schedule, :npq_leadership_autumn, cohort: new_cohort) }
+  let(:new_schedule_identifier) { new_schedule.identifier }
+
+  let(:params) do
+    {
+      lead_provider:,
+      participant:,
+      course_identifier:,
+
+      schedule_identifier: new_schedule_identifier,
+      cohort: nil,
+    }
+  end
+
+  subject { described_class.new(params) }
+
+  describe "validations" do
```

Comment:
There's a shared context for participant actions that should cover some of these tests:

https://github.com/DFE-Digital/npq-registration/blob/main/spec/support/shared_examples/participant_action_support.rb